### PR TITLE
Implement list-comprehension

### PR DESCRIPTION
**Pull request by FlorentinD**

### CODE REVIEW

File: okapi-ir/src/main/scala/org/opencypher/okapi/ir/impl/ExpressionConverter.scala [hunk 1]
```diff
@@ -62,18 +62,21 @@ final class ExpressionConverter(context: IRBuilderContext) {
     }
   }
 
-  def convert(e: ast.Expression): Expr = {
+  def convert(e: ast.Expression, lambdaField: Option[IRField] = None): Expr = {
```
[the remark this answers is not in the document]
Comment (FlorentinD) [reply okapi-ir/src/main/scala/org/opencypher/okapi/ir/impl/ExpressionConverter.scala]: I thought about it, but are we supporting list comprehensions inside list-comprehensions?

[the remark this answers is not in the document]
Comment (FlorentinD) [reply okapi-ir/src/main/scala/org/opencypher/okapi/ir/impl/ExpressionConverter.scala]: With a small change, nested list-comprehensions also working this way.
I would argue, that using a `Map` makes the code less readable, as there is only one active lambda-var at every point.

```
WITH [[1,2,3], [2,2,3], [3,4]] AS things
RETURN [x IN things | [n IN x WHERE n < 2]] AS value
```
In the nested list-comprehension, the lambda var `x` cannot be used in the `extract-expr` or in the `predicate-expr`.

[the remark this answers is not in the document]
Comment (FlorentinD) [reply okapi-ir/src/main/scala/org/opencypher/okapi/ir/impl/ExpressionConverter.scala]: just looking at `reduce()` ... using `Map[String, CypherType]` makes sense :smile:

[the remark this answers is not in the document]
Comment (DarthMax) [reply okapi-ir/src/main/scala/org/opencypher/okapi/ir/impl/ExpressionConverter.scala]: :D that is kind of what I had in mind. Using a map makes the concept more reusable in the future ;)

**Comment by FlorentinD** [comment]
@jjaderberg will do so on #916 as there are some more check marks :)

**Pull request merged**